contrib/babelfishpg_tsql/src/procedures.c

```
@@ -4290,3 +4304,528 @@ sp_reset_connection_internal(PG_FUNCTION_ARGS)
+ *
+ * The function temporarily elevates privileges to "bbf_role_admin" to ensure it has
+ * the necessary permissions to delete data from the table. It also sets the SQL
+ * dialect to TSQL to ensure proper deletion semantics.
+ */
+void 
+delete_xml_handle_entry(int document_id)
+{
+    Relation              relation;
+    ScanKeyData           skey[1];
+    TableScanDesc         scan;
+    HeapTuple             tuple;
+    bool                  found = false;
+    int                   curr_handle_counter;
+    int                   save_sec_context;
+    Oid                   save_userid;
+    int                   saved_dialect = sql_dialect;
+    MemoryContext         oldContext = NULL;
+    bool                  table_exists = false;
+    
+    /* Check for negative document ID */
+    if (document_id <= 0)
+    {
+        ereport(ERROR,
+                (errcode(ERRCODE_UNDEFINED_OBJECT),
+                 errmsg("Could not find prepared statement with handle %d", document_id)));
+    }
+
+    /* Calculate the handle counter */
+    curr_handle_counter = (document_id + 1) / 2;
+
+    /* Check if the handle is still active */
+    if (!bms_is_member(curr_handle_counter, active_xml_handles_counter))
+    {
+        ereport(ERROR,
+                (errcode(ERRCODE_UNDEFINED_OBJECT),
+                 errmsg("Could not find prepared statement with handle %d", document_id)));
+    }
+    
+    /* Check if the table exists using ENR lookup */
+    if(get_ENR_withoid(currentQueryEnv, xml_handle_temp_table_relid, ENR_TSQL_TEMP) != NULL)
+    {
+    	relation = relation_open(xml_handle_temp_table_relid, RowExclusiveLock);
+    	table_exists = true;
+    }
+
+    if (!table_exists)
+    {
+        /* Table doesn't exist, so the handle definitely doesn't exist */
+        ereport(ERROR,
+                (errcode(ERRCODE_UNDEFINED_OBJECT),
+                 errmsg("Could not find prepared statement with handle %d", document_id)));
```

please update this error message

---

If the table does not exist and user tries to remove a handle which does not exist , we should throw this error message as sql server gives the same error message when handle does not exist.